src/internal/operations/restore_test.go

```diff
@@ -218,38 +222,68 @@ func (suite *RestoreOpIntegrationSuite) SetupSuite() {
 	require.NotEmpty(t, bo.Results.BackupID)
 
 	suite.exchange = bupResults{
-		backupID: bo.Results.BackupID,
-		// Discount metadata files (3 paths, 3 deltas) as
+		selectorResourceOwners: users,
+		resourceOwner:          m365UserID,
+		backupID:               bo.Results.BackupID,
+		// Discount metadata collection files (1 delta
+		// and one prev path for each category) as
 		// they are not part of the data restored.
 		items: bo.Results.ItemsWritten - 6,
 	}
 
-	siteID := tester.M365SiteID(t)
-	sites := []string{siteID}
-	csel := selectors.NewSharePointBackup(sites)
-	csel.DiscreteOwner = siteID
-	csel.Include(csel.LibraryFolders(selectors.Any()))
+	var (
+		siteID = tester.M365SiteID(t)
+		sites  = []string{siteID}
+		spsel  = selectors.NewSharePointBackup(sites)
+	)
+
+	spsel.DiscreteOwner = siteID
+	// assume a folder name "test" exists in the drive.
+	// this is brittle, and requires us to backfill anytime
+	// the site under test changes, but also prevents explosive
+	// growth from re-backup/restore of restored files.
+	spsel.Include(spsel.LibraryFolders([]string{"test"}, selectors.PrefixMatch()))
 
 	bo, err = NewBackupOperation(
 		ctx,
 		control.Options{},
 		kw,
 		sw,
 		acct,
-		csel.Selector,
-		csel.Selector.DiscreteOwner,
+		spsel.Selector,
+		spsel.Selector.DiscreteOwner,
 		evmock.NewBus())
 	require.NoError(t, err, clues.ToCore(err))
 
+	// get the count of drives
+	m365, err := acct.M365Config()
+	require.NoError(t, err, clues.ToCore(err))
+
+	adpt, err := graph.CreateAdapter(
+		m365.AzureTenantID,
+		m365.AzureClientID,
+		m365.AzureClientSecret)
+	require.NoError(t, err, clues.ToCore(err))
+
+	service := graph.NewService(adpt)
+	spPgr := api.NewSiteDrivePager(service, siteID, []string{"id", "name"})
+
+	drives, err := api.GetAllDrives(ctx, spPgr, true, 3)
+	require.NoError(t, err, clues.ToCore(err))
+
 	err = bo.Run(ctx)
 	require.NoError(t, err, clues.ToCore(err))
 	require.NotEmpty(t, bo.Results.BackupID)
 
 	suite.sharepoint = bupResults{
-		backupID: bo.Results.BackupID,
-		// Discount metadata files (2 paths, 2 deltas) as
-		// they are not part of the data restored.
-		items: bo.Results.ItemsWritten - 4,
+		selectorResourceOwners: sites,
+		resourceOwner:          siteID,
+		backupID:               bo.Results.BackupID,
+		// Discount metadata files (1 delta, 1 prev path)
+		// assume only one folder, and therefore 1 dirmeta per drive
+		// assume only one file in each folder, and therefore 1 meta per drive
+		// all of these are not part of the data restored.
```

Comment:
The meta files are still used during restore (e.x. to source file names)
```suggestion
		// all of these are not directly part of the data restored.
```